### DISCUSSION TITLE
Frigate 13 Beta 7 Detecting Car with Car Mask enabled

### DISCUSSION THREAD

**Discussion by twister36** [blakeblackshear/frigate#9075] [Ask A Question]
Hi guys,

I've noticed that with Frigate 13 Beta 7 that the system is tracking objects even after a specific motion mask is applied where the Stationary object is sitting... 

For example, I have a car sitting in front of my house that I created a car motion mask so that it's not detected. However, Frigate continues to track the car and keep an active recording going, which causes large video files to be created. 

Here is the car with the mask:
![image](https://github.com/blakeblackshear/frigate/assets/114322877/334eaea6-b638-4af8-b52b-6d0ad77fc52d)

Here is the config for this camera:
![image](https://github.com/blakeblackshear/frigate/assets/114322877/26f5cec9-9128-40de-9cc8-514942ec8e2d)
(Note: the section with comment "#YUKON PARKING STREET", this is the mask for the object tracking for a car)

From the events page you can see an active "in-progress"  recording for the car in which I have a car mask for. The system has been recording the car object for over 35+mins and will keep going:
![image](https://github.com/blakeblackshear/frigate/assets/114322877/6a850461-8a62-49a5-87bc-0d79e716f635)

Any idea why the system is still tracking this car object and why it keeps recording it.

**Comment by NickM-27** [comment]
Motion masks do not stop objects from being detected like it says in the docs. 

Also like it says, the event being in progress does not mean it is being constantly recorded

**Reply by NickM-27** [reply]
You should use required zones to restrict events to important areas

**Reply by twister36** [reply]
I'm not creating a "motion mask", I created an "object filter mask", which should tell the system not to look for a "car" in that area.

Can you clarify, "Also like it says, the event being in progress does not mean it is being constantly recorded". I can see that it's created 82 video files for the "driveway" for this object in the last 30 mins.

**Reply by NickM-27** [reply]
Object masks filter objects based on the bottom center of the bounding box. In your example the bottom center is outside the mask. In any case a required zone should be used to mask out uninteresting objects

**Reply by twister36** [reply]
@NickM-27 - I'm not sure I understand what you mean that bottom center is outside of the mask?? Wouldn't the orange line be bottom center?
![image](https://github.com/blakeblackshear/frigate/assets/114322877/41b14017-6108-4d97-88cd-0fd249e2a4ff)

**Reply by NickM-27** [reply]
Bottom center is a single pixel on the red box. It is outside the mask, that is the only position that matters when deciding if an object is filtered by a mask or not.

**Reply by twister36** [reply]
@NickM-27 - Ok, let me clarify so that I understand... 

Are you saying:
A) that any part of the red box is outside of the mask will trigger the object to be evaluated?
or 
B) that if the bottom-center pixel is outside the mask will trigger the object to be evaluated?

**Reply by NickM-27** [reply]
This is covered here https://docs.frigate.video/configuration/masks

a) no
b) yes

**Reply by twister36** [reply]
@NickM-27 Thank you for that link, it explains allot. I'm not sure if you're just very helpful or apart of the Frigate development team, but it appears that on the docs for Frigate Configuration file, "required_zones" is not mentioned as an option under the Camera node:
https://deploy-preview-6262--frigate-docs.netlify.app/configuration/

However, I do see it listed on this page:
https://deploy-preview-6262--frigate-docs.netlify.app/guides/stationary_objects/

Again, thanks for your help.

**Reply by NickM-27** [reply]
I do contribute to the frigate codebase as a dev. In the config reference docs each section shows "can be overridden at the camera level" which means all options can be set at camera level.

**Reply by twister36** [reply]
Previously I didn't use recordings because my Raspberry Pi couldn't keep up with it. However since my hardware has been upgraded I'm full of "recording" related questions. I owe you a beer or at this point several!
